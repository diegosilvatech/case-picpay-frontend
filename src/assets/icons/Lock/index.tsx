import IconBase, { IconBaseProps } from 'components/IconBase';

export type IconType = Pick<IconBaseProps, 'color'>;

export const LockIcon = ({ color }: IconType) => (
  <IconBase color={color}>
    <svg
      xmlns="http://www.w3.org/2000/svg"
      height="24px"
      viewBox="0 0 24 24"
      width="24px"
      aria-hidden="true"
      fill="none"
      stroke="currentColor"
      data-testid="lock-icon"
    >
      <rect width="18" height="11" x="3" y="11" rx="2" ry="2"></rect>
      <path d="M7 11V7a5 5 0 0110 0v4"></path>
    </svg>
  </IconBase>
);
