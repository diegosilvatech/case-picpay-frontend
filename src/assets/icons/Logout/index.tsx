import IconBase, { IconBaseProps } from 'components/IconBase';

export type IconType = Pick<IconBaseProps, 'color'>;

export const LogoutIcon = ({ color }: IconType) => (
  <IconBase color={color}>
    <svg
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 24 24"
      height="24"
      width="24"
      focusable="false"
      role="img"
      fill="none"
      stroke="currentColor"
      data-testid="logout-icon"
    >
      <path d="M9 21H5a2 2 0 01-2-2V5a2 2 0 012-2h4"></path>
      <polyline points="16 17 21 12 16 7"></polyline>
      <line x1="21" x2="9" y1="12" y2="12"></line>
    </svg>
  </IconBase>
);
