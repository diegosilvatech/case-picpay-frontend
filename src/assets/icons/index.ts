/* PLOP_INJECT_EXPORT */
export { LogoutIcon } from './Logout';
export { LockIcon } from './Lock';
export { MailIcon } from './Mail';
export { ArrowDownIcon } from './ArrowDown';
