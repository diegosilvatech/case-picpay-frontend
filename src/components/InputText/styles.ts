import styled, { css, DefaultTheme } from 'styled-components';

import { InputTextProps } from '.';

type IconPositionProps = Pick<InputTextProps, 'iconPosition'>;
type WrapperProps = Pick<InputTextProps, 'disabled'> & {
  error?: boolean;
};

const textFieldModifiers = {
  disabled: (theme: DefaultTheme) => css`
    ${Label},
    ${Input},
    ${InputWrapper} {
      cursor: not-allowed;
      color: ${theme.colors.gray};

      &::placeholder {
        color: currentColor;
      }
    }
    ${IconWrapper} {
      svg {
        color: ${theme.colors.gray};
      }
    }
  `,
  error: (theme: DefaultTheme) => css`
    ${InputWrapper} {
      border: 0.25rem solid ${theme.colors.error};

      &:focus-within {
        box-shadow: 0 0 0.25rem ${theme.colors.error};
      }
    }
    ${Input},
    ${Label} {
      color: ${theme.colors.error};
    }
    ${IconWrapper} {
      svg {
        color: ${theme.colors.error};
      }
    }
  `
};

export const Wrapper = styled.div<WrapperProps>`
  ${({ theme, disabled, error }) => css`
    ${disabled && textFieldModifiers.disabled(theme)}
    ${error && textFieldModifiers.error(theme)}
  `}
`;

export const InputWrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    background: ${theme.colors.white};
    border-radius: ${theme.border.radius.small};
    padding: 0 ${theme.spacings.small};
    border: 0.25rem solid;

    &:focus-within {
      box-shadow: 0 0 0.25rem ${theme.colors.primary};
    }
  `}
`;

export const Input = styled.input<IconPositionProps>`
  ${({ theme, iconPosition }) => css`
    color: ${theme.colors.black};
    font-family: ${theme.font.family};
    padding: ${theme.spacings.medium} 0;
    padding-${iconPosition}: ${theme.spacings.small};
    background: transparent;
    border: none;
    outline: none;
    width: 100%;
  `}
`;

export const LabelWrapper = styled.div`
  ${({ theme }) => css`
    margin-bottom: ${theme.spacings.extraSmall};
  `}
`;

export const Label = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.default};
    font-weight: ${theme.font.weights.regular};
    color: ${theme.colors.black};
  `}
`;

export const IconWrapper = styled.div<IconPositionProps>`
  ${({ iconPosition }) => css`
    display: flex;
    align-items: center;
    order: ${iconPosition === 'right' ? 1 : 0};
  `}
`;

export const Error = styled.p`
  ${({ theme }) => css`
    color: ${theme.colors.error};
    font-size: 1.6rem;
  `}
`;
