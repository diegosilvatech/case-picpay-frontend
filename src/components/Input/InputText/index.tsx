import { InputHTMLAttributes } from 'react';

import * as s from './styles';

export type InputTextProps = {
  label?: string;
  icon?: React.ReactNode;
  iconPosition?: 'left' | 'right';
  disabled?: boolean;
  error?: string;
  value?: string;
} & InputHTMLAttributes<HTMLInputElement>;

const InputText = ({
  disabled = false,
  error,
  icon,
  iconPosition = 'left',
  label,
  name,
  value,
  onChange,
  ...props
}: InputTextProps) => {
  return (
    <s.Wrapper
      aria-label="InputText component"
      error={!!error}
      disabled={disabled}
    >
      {!!label && (
        <s.LabelWrapper>
          <s.Label htmlFor={name}>{label}</s.Label>
        </s.LabelWrapper>
      )}
      <s.InputWrapper>
        {!!icon && (
          <s.IconWrapper iconPosition={iconPosition}>{icon}</s.IconWrapper>
        )}
        <s.InputText
          type="text"
          name={name}
          onChange={onChange}
          value={value}
          iconPosition={iconPosition}
          disabled={disabled}
          {...(label ? { id: name } : {})}
          {...props}
        />
      </s.InputWrapper>
      {!!error && <s.Error>{error}</s.Error>}
    </s.Wrapper>
  );
};

export default InputText;
