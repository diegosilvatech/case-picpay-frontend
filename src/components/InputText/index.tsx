import { InputHTMLAttributes, useState } from 'react';

import * as s from './styles';

export type InputTextProps = {
  label?: string;
  initialValue?: string;
  icon?: React.ReactNode;
  iconPosition?: 'left' | 'right';
  disabled?: boolean;
  error?: string;
  onInput?: (value: string) => void;
} & InputHTMLAttributes<HTMLInputElement>;

const InputText = ({
  disabled = false,
  error,
  icon,
  iconPosition = 'left',
  initialValue = '',
  label,
  name,
  onInput,
  ...props
}: InputTextProps) => {
  const [value, setValue] = useState(initialValue);
  const onChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    const newValue = event.currentTarget.value;
    setValue(newValue);

    !!onInput && onInput(newValue);
  };

  return (
    <s.Wrapper
      aria-label="InputText component"
      error={!!error}
      disabled={disabled}
    >
      {!!label && (
        <s.LabelWrapper>
          <s.Label htmlFor={name}>{label}</s.Label>
        </s.LabelWrapper>
      )}
      <s.InputWrapper>
        {!!icon && (
          <s.IconWrapper iconPosition={iconPosition}>{icon}</s.IconWrapper>
        )}
        <s.Input
          type="text"
          name={name}
          onChange={onChange}
          value={value}
          iconPosition={iconPosition}
          disabled={disabled}
          {...(label ? { id: name } : {})}
          {...props}
        />
      </s.InputWrapper>
      {!!error && <s.Error>{error}</s.Error>}
    </s.Wrapper>
  );
};

export default InputText;
