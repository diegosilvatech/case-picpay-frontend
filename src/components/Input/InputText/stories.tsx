import { Story, Meta } from '@storybook/react/types-6-0';

import { MailIcon } from 'assets/icons';

import InputText, { InputTextProps } from '.';

export default {
  title: 'Molecules/Form/InputText',
  component: InputText,
  decorators: [
    (Story) => (
      <div style={{ maxWidth: '40rem', padding: 15 }}>
        <Story />
      </div>
    )
  ],
  parameters: {
    jest: ['Input.test.tsx']
  },
  args: {
    name: 'email',
    label: 'e-mail',
    placeholder: 'diego.silva@picpay.com',
    icon: <MailIcon />,
    iconPosition: 'left',
    disabled: false
  },
  argTypes: {
    onChange: { action: 'changed' }
  }
} as Meta;

export const InputDefault: Story<InputTextProps> = (args) => (
  <InputText {...args} />
);

export const InputWithError: Story<InputTextProps> = (args) => (
  <InputText {...args} />
);

InputWithError.args = {
  error: '...something is wrong'
};
