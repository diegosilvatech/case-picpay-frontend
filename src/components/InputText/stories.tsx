import { Story, Meta } from '@storybook/react/types-6-0';

import { MailIcon } from 'assets/icons';

import InputText, { InputTextProps } from '.';

export default {
  title: 'Molecules/Form/InputText',
  component: InputText,
  decorators: [
    (Story) => (
      <div style={{ maxWidth: '40rem', padding: 15 }}>
        <Story />
      </div>
    )
  ],
  parameters: {
    jest: ['InputText.test.tsx']
  },
  args: {
    name: 'email',
    label: 'e-mail',
    placeholder: 'diego.silva@picpay.com',
    icon: <MailIcon />,
    iconPosition: 'left',
    disabled: false
  },
  argTypes: {
    onChange: { action: 'changed' }
  }
} as Meta;

export const InputTextDefault: Story<InputTextProps> = (args) => (
  <InputText {...args} />
);

export const InputTextWithError: Story<InputTextProps> = (args) => (
  <InputText {...args} />
);

InputTextWithError.args = {
  error: '...something is wrong'
};
