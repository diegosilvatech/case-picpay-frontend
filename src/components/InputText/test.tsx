import { screen } from '@testing-library/react';
import { renderWithTheme } from 'core/helpers/tests/renderWithTheme';

import userEvent from '@testing-library/user-event';

import { MailIcon } from 'assets/icons';

import InputText from '.';

describe('<InputText />', () => {
  it('should render with label', () => {
    renderWithTheme(<InputText label="label" name="field" />);

    expect(screen.getByLabelText('label')).toBeInTheDocument();
  });

  it('should render without Label', () => {
    renderWithTheme(<InputText />);

    expect(screen.queryByLabelText('label')).not.toBeInTheDocument();
  });

  it('should render with placeholder', () => {
    renderWithTheme(<InputText placeholder="some placeholder" />);

    expect(screen.getByPlaceholderText('some placeholder')).toBeInTheDocument();
  });

  it('should accessible by tab key', () => {
    renderWithTheme(<InputText label="inputText" name="inputText" />);

    const input = screen.getByLabelText('inputText');
    expect(document.body).toHaveFocus();

    userEvent.tab();
    expect(input).toHaveFocus();
  });

  it('should render with Icon', () => {
    renderWithTheme(<InputText icon={<MailIcon />} />);

    expect(screen.getByTestId('mail-icon')).toBeInTheDocument();
  });

  it('should render with error', () => {
    const { container } = renderWithTheme(
      <InputText
        icon={<MailIcon data-testid="icon" />}
        label="inputText"
        name="inputText"
        error="error message"
      />
    );

    expect(screen.getByText('error message')).toBeInTheDocument();
    expect(screen.getByText('error message')).toHaveStyle({ color: '#DD2C00' });
    expect(container.firstChild).toMatchSnapshot();
  });
});
