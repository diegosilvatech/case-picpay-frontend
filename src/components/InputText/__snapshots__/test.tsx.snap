// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputText /> should render with error 1`] = `
.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #000000;
}

.c0 .c3 {
  border: 0.25rem solid #DD2C00;
}

.c0 .c3:focus-within {
  box-shadow: 0 0 0.25rem #DD2C00;
}

.c0 .c8,
.c0 .c1 {
  color: #DD2C00;
}

.c0 .c5 svg {
  color: #DD2C00;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #FFFFFF;
  border-radius: 0.5rem;
  padding: 0 1rem;
  border: 0.25rem solid;
}

.c4:focus-within {
  box-shadow: 0 0 0.25rem #11c76f;
}

.c9 {
  color: #000000;
  font-family: Montserrat,Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  padding: 2rem 0;
  padding-left: 1rem;
  background: transparent;
  border: none;
  outline: none;
  width: 100%;
}

.c2 {
  font-size: 1.75rem;
  font-weight: 400;
  color: #000000;
  cursor: pointer;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-order: 0;
  -ms-flex-order: 0;
  order: 0;
}

.c10 {
  color: #DD2C00;
  font-size: 1.6rem;
}

<div
  aria-label="InputText component"
  class="c0"
>
  <label
    class="c1 c2"
    for="inputText"
  >
    inputText
  </label>
  <div
    class="c3 c4"
  >
    <div
      class="c5 c6"
    >
      <div
        aria-label="IconBase component"
        class="c7"
      >
        <svg
          data-testid="mail-icon"
          fill="none"
          height="24px"
          stroke="currentColor"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M4 4h16c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H4c-1.1 0-2-.9-2-2V6c0-1.1.9-2 2-2z"
          />
          <polyline
            points="22,6 12,13 2,6"
          />
        </svg>
      </div>
    </div>
    <input
      class="c8 c9"
      id="inputText"
      name="inputText"
      type="text"
      value=""
    />
  </div>
  <p
    class="c10"
  >
    error message
  </p>
</div>
`;
